tests/samplers_tests/test_nsgaii.py

```diff
@@ -157,6 +155,25 @@ def constraints_func(trial: FrozenTrial) -> Sequence[float]:
             assert _nan_equal(x, y)
 
 
+def test_constraints_func_nan() -> None:
+    n_trials = 4
+    n_objectives = 2
+
+    def constraints_func(_: FrozenTrial) -> Sequence[float]:
+        return (float("nan"),)
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
+        sampler = NSGAIISampler(population_size=2, constraints_func=constraints_func)
+
+    study = optuna.create_study(directions=["minimize"] * n_objectives, sampler=sampler)
+    with pytest.raises(ValueError):
+        study.optimize(
+            lambda t: [t.suggest_float(f"x{i}", 0, 1) for i in range(n_objectives)],
+            n_trials=n_trials,
+        )
+
```

Comment:
Could you check that the `study` is healthy? I mean testing the trial's `state` and `value`.

Reply:
Thanks, fixed.